src/main/scala/com/github/codelionx/dodo/actors/Worker.scala

```diff
@@ -0,0 +1,78 @@
+package com.github.codelionx.dodo.actors
+
+import akka.actor.{Actor, ActorLogging, ActorRef, Props}
+import com.github.codelionx.dodo.actors.DataHolder.DataRef
+import com.github.codelionx.dodo.discovery.{CandidateGenerator, Pruning}
+import com.github.codelionx.dodo.types.TypedColumn
+
+import scala.collection.immutable.Queue
+
+
+object Worker {
+
+  val name = "worker"
+
+  def props(): Props = Props[Worker]
+
+  case object GetTask
+
+  case class CheckForEquivalency(oeToCheck: (Int, Int))
+
+  case class OrderEquivalent(oe: (Int, Int), isOrderEquiv: Boolean)
+
+  case class CheckForOD(odToCheck: (List[Int], List[Int]), reducedColumns: Set[Int])
+
+  case class ODsToCheck(parentOD: (List[Int], List[Int]), newODs: Queue[(List[Int], List[Int])])
+
+  case class ODFound(od: (List[Int], List[Int]))
+
+}
+
+
+class Worker extends Actor with ActorLogging with Pruning with CandidateGenerator{
+
+  import Worker._
+
+
+  override def preStart(): Unit = {
+    log.info(s"Starting $name")
+    Reaper.watchWithDefault(self)
+  }
+
+  override def postStop(): Unit =
+    log.info(s"Stopping $name")
+
+  override def receive: Receive = uninitialized
+
+  def uninitialized: Receive = {
+    case DataRef(table) =>
+      context.become(workReady(table))
+      sender ! GetTask
+    case _ => log.info("Unknown message received")
+  }
+
+  def workReady(table: Array[TypedColumn[Any]]): Receive = {
+    case CheckForEquivalency(oeToCheck) =>
+      sender ! OrderEquivalent(oeToCheck, checkOrderEquivalent(table(oeToCheck._1), table(oeToCheck._2)))
+      sender ! GetTask
+    case CheckForOD(odCandidate, reducedColumns) =>
```

Comment:
This message handler does only represent the `generateNextLevel` method (ALGORITHM3 in the paper) and misses the order compatibility dependency check (OCD) from the overall algorithm (ALGORITHM1 L8 - L11)